app/controllers/api/assignments_controller.rb

```diff
@@ -5,6 +5,8 @@ class AssignmentsController < MainApiController
     include AutomatedTestsHelper
     include SubmissionsHelper
 
+    authorize :course, through: :current_course
```

Comment:
It's not necessary to pass this into the policy. The policy already has access to `role`, which is computed through the current course. So it should be sufficient to check that `role` is not `nil` in the policy.